### ISSUE TITLE
Aggregation: Ability to add metadata

### ISSUE THREAD

**Issue by leflings** [ruflin/Elastica#1677]
Hi

First of all, thank you for a very nice library, which I've enjoyed using (with/through [FOSElasticaBundle](https://github.com/FriendsOfSymfony/FOSElasticaBundle))

ElasticSearch has the ability to add _[metadata](https://www.elastic.co/guide/en/elasticsearch/reference/6.8/agg-metadata.html)_ to aggregations, a feature I've found to be wanting while making a faceted search solution.

Is it possible to emulate this with the current API (I'm using version 6.1.1)?

If not, could this be added to the Aggregation API? Hopefully in a way where it lands in a 6.X version (the bundle doesn't yet support versions 7.X)

**Comment by ruflin** [comment]
Just learned a new feature in ES, thanks! My guess would be that `setParam` could be  used on each aggregation to add it. I didn't try it out but as all Aggregations inherit `Params`, this should work: https://github.com/ruflin/Elastica/blob/master/lib/Elastica/Aggregation/AbstractAggregation.php#L9 

Happy to get native support for it, interested to contribute 😇

**Comment by leflings** [comment]
I tried `setParam` but it unfortunately doesn't generate a valid aggregation.

The following code:
```
$termsAgg = new \Elastica\Aggregation\Terms('titles');
$termsAgg->setField('title');
$termsAgg->setParam('meta', ['color' => 'blue']);

$query = new \Elastica\Query();
$query->addAggregation($termsAgg);
```

Generates the following query:
```
{
    "aggs": {
        "titles": {
            "terms": {
                "field": "title",
                "meta": {
                    "color": "blue"
                }
            }
        }
    },
    "query": {
        "match_all": {}
    }
}
```

Which generates the following response:
```
{
  "error": {
    "root_cause": [
      {
        "type": "x_content_parse_exception",
        "reason": "[6:17] [terms] unknown field [meta], parser not found"
      }
    ],
    "type": "x_content_parse_exception",
    "reason": "[6:17] [terms] unknown field [meta], parser not found"
  },
  "status": 400
}
```

The `meta` object needs to be output "one level higher" / at the same level as the aggregation type. See [Structuring Aggregations](https://www.elastic.co/guide/en/elasticsearch/reference/6.8/search-aggregations.html#_structuring_aggregations).

I've locally implemented `setMeta(array)` and `clearMeta()` on `AbstractAggregation` and handled the new field in `toArray()`. It works just fine. I'll see about making a proper pull request later in the day.

**Comment by leflings** [comment]
I've made a pull request (#1678) towards the `6.x` branch. If this is the wrong way around and you would  rather have a request towards the master branch, that can then be backported, say so. I guess my thinking was that this commit could be "forwardported" :)

_Disclaimer: I could not get the docker environment running properly for the tests to complete (neither when trying the same commit rebased to the master branch). It kept waiting on the ElasticSearch server to be ready, which it never became. The `lint` and `check-style` steps worked. I've run both tests manually elsewhere, so I'm pretty certain the would pass in a working environment._

**Comment by ruflin** [comment]
@leflings Thank you! Yes, I would prefer to have it in master first and backport it from there if possible.

**Comment by leflings** [comment]
@ruflin Opened a new pull request (#1680) based on `master` and using existing `_rawParams`

**Comment by ruflin** [comment]
Great, thanks. Already merged. Do you want to open the backport PR from there?
